lib/Sema/SemaDecl.cpp

```diff
@@ -8180,6 +8180,7 @@ void Sema::CheckMain(FunctionDecl* FD, const DeclSpec& DS) {
           Context.hasSameType(QualType(qs.strip(PT->getPointeeType()), 0),
                               Context.CharTy)) {
         qs.removeConst();
+        qs.removeAddressSpace();
```

Comment:
There doesn't seem to be a way to query then default address space from ASTContext. Presumably the value we want to compare against is the AllocaAS in the target's DataLayout, but I can't find a way to reach that from here.

Perhaps a neater approach is to preserve addressspace qualifiers in the type comparison a couple of lines up, so hasSameType will fail for us and we don't need to sellotape on an extra check?